core/src/com/unciv/logic/civilization/QuestManager.kt

```diff
@@ -339,6 +342,28 @@ class QuestManager {
 
         return null
     }
+
+    /**
+     * Returns a Great Person [BaseUnit] that is not owned by both the [challenger] and the [civInfo]
+     */
+    private fun getGreatPersonForQuest(challenger: CivilizationInfo): BaseUnit? {
+        val ruleSet = civInfo.gameInfo.ruleSet
+
+        val challengerGreatPeople = challenger.getCivGreatPeople().map { it.baseUnit.getReplacedUnit(ruleSet) }
+        val cityStateGreatPeople = civInfo.getCivGreatPeople().map { it.baseUnit.getReplacedUnit(ruleSet) }
+
+        val greatPeople = ruleSet.units.values
```

Comment:
. asSequence()

Reply:
To extract a random element the container must be a list-like. I have to create a list from the sequence after all the filters.